src/ToricVarieties/NormalToricVarieties/attributes.jl

```diff
@@ -313,6 +313,131 @@ end
+            col += 1
+        end
+    end
+    
+    # compute the matrix for differences
+    map_for_difference_of_elements = zero_matrix(ZZ, s, s-number_of_rays)
+    row = 1
+    col = 1
+    for i in 1:number_of_rays
+        ncol = number_ray_is_part_of_max_cones[i]-1
+        map_for_difference_of_elements[row, col:(col+ncol-1)] = fill(fmpz(1), ncol)
+        map_for_difference_of_elements[(row+1):(row+ncol), col:(col+ncol-1)] = -identity_matrix(ZZ, ncol)
+        row += ncol + 1
+        col += ncol
+    end
+    
+    # compute the matrix for mapping to torusinvariant Weil divisors
+    map_to_weil_divisors = zero_matrix(ZZ, number_of_cones * rc, rank(torusinvariant_divisor_group(v)))
+    for i in 1:number_of_rays
+        map_to_weil_divisors[(cones_ray_is_part_of[i][1]-1)*rc+1:cones_ray_is_part_of[i][1]*rc, i] = [fmpz(-c) for c in rays[i,:]]
+    end
+    
+    # compute the total map
+    mapping_matrix = map_for_scalar_products * map_for_difference_of_elements
+    source = abelian_group(zeros(Int, nrows(mapping_matrix)))
+    target = abelian_group(zeros(Int, ncols(mapping_matrix)))
+    total_map = hom(source, target, mapping_matrix)
+    
+    # identify the cartier_data_group and its embedding
+    ker = kernel(total_map)
+    cartier_data_group = snf(ker[1])[1]
+    cartier_data_group_embedding = hom(cartier_data_group, source, ker[1].snf_map.map * ker[2].map)
+    
+    # construct the embedding of the Cartier divisor group into the group of torusinvariant divisors
+    return cartier_divisor_group_embedding = image(hom(cartier_data_group, torusinvariant_divisor_group(v), cartier_data_group_embedding.map * map_to_weil_divisors))[2]
+end
+export map_from_cartier_divisor_group_to_torus_invariant_divisor_group
+
+@doc Markdown.doc"""
+    cartier_divisor_group(v::AbstractNormalToricVariety)
+
+Computes the Cartier divisor group of an abstract normal toric variety `v`.
+
+# Examples
+```jdoctest
+julia> p2 = toric_projective_space(2)
+A normal toric variety corresponding to a polyhedral fan in ambient dimension 2
+
+julia> cartier_divisor_group(p2)
+GrpAb: Z^3
+```
+"""
+function cartier_divisor_group(v::AbstractNormalToricVariety)
+    return domain(map_from_cartier_divisor_group_to_torus_invariant_divisor_group(v))
+end
+export cartier_divisor_group
+
+
+@doc Markdown.doc"""
+    picard_group(v::AbstractNormalToricVariety)
+
+Computes the Picard group of an abstract normal toric variety `v`.
+
+# Examples
+```jdoctest
+julia> p2 = toric_projective_space(2)
+A normal toric variety corresponding to a polyhedral fan in ambient dimension 2
+
+julia> picard_group(p2)
+GrpAb: Z
+```
+"""
+function picard_group(v::AbstractNormalToricVariety)
+    map1 = map_from_cartier_divisor_group_to_torus_invariant_divisor_group(v)
+    map2 = map_from_weil_divisors_to_class_group(v)
+    return snf(image(hom(domain(map1), codomain(map2), map1.map * map2.map))[1])[1]
```

Comment:
This relies on an implementation detail of those maps, i.e. the code is not actually generic. Can't you use `compose` resp `*` on map1 and map2 directly, which would also give shorter code